crates/astria-conductor/src/sequencer/mod.rs

```diff
@@ -285,22 +289,58 @@ async fn subscribe_new_blocks(
     .map_err(|_| ResubscriptionError::BackoffFailed)
 }
 
-fn forward_pending_block(
+fn forward_pending_block_or_resync(
+    expected_height: u32,
     executor_tx: executor::Sender,
+    pool: Pool<ClientProvider>,
+    pending_blocks: &mut FuturesOrdered<Ready<Result<SequencerBlockData, NewBlockStreamError>>>,
+    sync: &mut future::Fuse<Pin<Box<dyn Future<Output = eyre::Result<()>> + Send>>>,
     res: Result<SequencerBlockData, NewBlockStreamError>,
-) -> eyre::Result<()> {
+) -> eyre::Result<u32> {
```

Comment:
can you add a comment explaining the return value?

Reply:
it also doesn't look like the value is used anywhere?

Reply:
Good catch, that was important to include. I am now using the return value and added a mention in the doc comment on what it returns.